sdk/src/main/java/io/opentdf/platform/sdk/KASClient.java

```diff
@@ -169,19 +165,26 @@ static class NanoTDFRewrapRequestBody {
     @Override
     public byte[] unwrap(Manifest.KeyAccess keyAccess, String policy,  KeyType sessionKeyType) {
         ECKeyPair ecKeyPair = null;
+        
+        if (sessionKeyType.isEc()) {
+            var curveName = sessionKeyType.getCurveName();
+            ecKeyPair = new ECKeyPair(curveName, ECKeyPair.ECAlgorithm.ECDH);
+            clientPublicKey = ecKeyPair.publicKeyInPEMFormat();
+        } else {
+            // Initialize the RSA key pair only once and reuse it for future unwrap operations
+            if (decryptor == null) {
+                var encryptionKeypair = CryptoUtils.generateRSAKeypair();
+                decryptor = new AsymDecryption(encryptionKeypair.getPrivate());
+                clientPublicKey = CryptoUtils.getRSAPublicKeyPEM(encryptionKeypair.getPublic());
+            }
```

Comment:
maybe throw this in a synchronized method since this makes this not threadsafe. I have no idea if this can be called from multiple threads and we should probably specify what we support but it seems like a decent idea.